### PR TITLE
stack update

### PR DESCRIPTION
**Pull request by aherrmann-da**
- Updates to latest rules_haskell
- Adds a patch for compatibility (optional cabal haddock) which can be removed once the upstream PR is merged
- The new rules_haskell version should fix the hackage security lock issues we've been seeing on macOS CI. The corresponding workaround on Windows should no longer be required either.

### Pull Request Checklist
- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags, if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/haskell-optional-cabal-haddock.patch [hunk 1]
```diff
@@ -0,0 +1,260 @@
+From 0d20e37429257b97d2f5387a32f2062b90426a13 Mon Sep 17 00:00:00 2001
```
[the remark this answers is not in the document]
Comment (aherrmann-da) [reply bazel_tools/haskell-optional-cabal-haddock.patch]: There's no global flag for this, but it's not really necessary. With the regular `haskell_library|binary|test` rules the haddocks are only generated if explicitly demanded using a `haskell_doc` target. The Cabal rules are different, in that they run the full build in a single action, and therefore have to include haddock generation in that step if at all.

[the remark this answers is not in the document]
Comment (cocreature) [reply bazel_tools/haskell-optional-cabal-haddock.patch]: Thanks for the explanation! I guess our external deps change sufficiently rarely that this is not something to worry about.

[the remark this answers is not in the document]
Comment (aherrmann-da) [reply bazel_tools/haskell-optional-cabal-haddock.patch]: This PR does disable Haddock generation for all external dependencies defined via `stack_snapshot` and `haskell_cabal_library` in `bazel-haskell-deps.bzl`.

[the remark this answers is not in the document]
Comment (cocreature) [reply bazel_tools/haskell-optional-cabal-haddock.patch]: Ohhh, I missed the `haddock = False` in `stack_snapshot`. Perfect, thank you!

**Pull request merged**